### PR TITLE
feat: enhance cross-chain swap status tracking with comprehensive provider support

### LINKED ISSUE #629: Implement transaction status for cross chain providers

### PR DESCRIPTION
**Pull request by 0xh3rman**
## Summary
This PR significantly enhances the swap status tracking system by implementing comprehensive cross-chain transaction tracking across all swap providers (ThorChain, Chainflip, Across, and Mayan).

#### 🏗️ Architecture Improvements
- **Extracted Swapper trait** to dedicated `swapper_trait.rs` module for better organization
- **Enhanced return type** from simple `bool` to rich `SwapperSwapResult` containing detailed transaction information
- **Added reusable helper method** `get_onchain_swap_status()` for consistent OnChain provider handling

#### ⛓️ Cross-Chain Provider Implementations

**ThorChain Provider:**
- ✅ Comprehensive memo parsing to extract destination chains from transaction memos
- ✅ Smart `out_hashes` processing to find first non-zero destination transaction hash
- ✅ RuneScan integration for proper block explorer URLs
- ✅ Extensive unit tests for memo parsing and chain name mapping functions
- ✅ Support for both long names (ETH, BSC, AVAX) and short names (e, s, a) in memo parsing

**Chainflip Provider:**
- ✅ Complete status tracking with COMPLETED/FAILED/PENDING state mapping
- ✅ Automatic chain name conversion using `Chain::from_str()` with lowercase normalization
- ✅ ChainflipScan integration for specialized block explorer URLs
- ✅ Proper destination chain and transaction extraction from swap egress data

**Across Provider:**
- ✅ Enhanced deposit status tracking with filled/refunded/pending states
- ✅ SocketScan integration for cross-chain transaction URLs
- ✅ Intelligent destination chain resolution from chain IDs
- ✅ Proper transaction hash mapping based on swap completion status

**Mayan Provider:**
- ✅ Comprehensive client status mapping (COMPLETED/IN_PROGRESS/REFUNDED)
- ✅ Chain name mapping with special handling for BSC→SmartChain and Avalanche→AvalancheC
- ✅ MayanScan integration for transaction tracking
- ✅ Proper fulfill/refund transaction hash extraction

#### 🔄 Consistency Improvements
- **Unified Status Pattern**: All providers now use dedicated `swap_status()` methods returning `SwapStatus` enum
- **Consistent Refund Handling**: All providers set `to_tx_hash` to `None` for refunded transactions
- **Clean Code Patterns**: Removed redundant helper methods in favor of direct status mapping
- **Organized Imports**: Grouped import statements for better code organization

#### 🧪 Testing
- ✅ Comprehensive unit tests for ThorChain memo parsing
- ✅ Chain name mapping validation tests
- ✅ Status conversion testing across all providers
- ✅ Edge case handling for various transaction states

#### ThorChain Memo Parsing
The implementation supports ThorChain's memo format: `=:ASSET:ADDRESS:LIMIT/INTERVAL/QUANTITY:AFFILIATE:BPS`
- Handles both token swaps (`ETH.USDT`) and native asset swaps (`e`, `s`, etc.)
- Uses strum's `EnumIter` for automatic enum iteration
- Leverages existing `long_name()` and `short_name()` methods for DRY compliance

#### Cross-Chain Transaction Tracking
- **Source Chain**: Always the chain where the original transaction occurred
- **Destination Chain**: Parsed from provider-specific data (memo, API response, chain ID)
- **Transaction Hashes**: Proper mapping of source and destination transaction hashes
- **Explorer URLs**: Provider-specific block explorer integration

#### Status State Management
```rust
pub enum SwapStatus {
    Pending,    // Transaction in progress
    Completed,  // Successfully completed cross-chain swap
    Failed,     // Failed with specific error
    Refunded,   // Transaction was refunded to source chain
}
```

### Benefits
1. **Rich Transaction Information**: Detailed swap results with source/destination chains and transaction hashes
2. **Consistent API**: Unified interface across all swap providers
3. **Better User Experience**: Users can track their cross-chain swaps with proper block explorer links
4. **Maintainable Code**: Clean patterns and consistent status handling across providers
5. **Comprehensive Testing**: Robust test coverage for parsing and status detection logic

### Backward Compatibility
- ✅ All existing functionality preserved
- ✅ OnChain providers continue to work with fallback implementation
- ✅ No breaking changes to public APIs

Fixes #629

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: gemstone/src/swapper/thorchain/chain.rs [hunk 1]
```diff
@@ -94,4 +95,154 @@ impl THORChainName {
...
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_from_symbol() {
```
Comment (gemcoder21) [inline gemstone/src/swapper/thorchain/chain.rs:134]: useless unit test, why it's needed?

File: gemstone/src/swapper/thorchain/chain.rs [hunk 2]
```diff
@@ -94,4 +95,154 @@ impl THORChainName {
...
+    /// Supports both long names (ETH, BSC, AVAX) and short names (e, s, a)
+    pub fn from_symbol(symbol: &str) -> Option<THORChainName> {
+        // Use strum's EnumIter to automatically iterate over all enum variants
+        THORChainName::iter().find(|variant| variant.long_name() == symbol || variant.short_name() == symbol)
+    }
+
+    /// Parse THORChain memo to extract destination chain
+    /// Memo format: =:ASSET:ADDRESS:LIMIT/INTERVAL/QUANTITY:AFFILIATE:BPS
+    /// Example: "=:ETH.USDT:0x858734a6353C9921a78fB3c937c8E20Ba6f36902:1635978e6/1/0:-_/ll:0/150"
+    pub fn parse_dest_from_memo(memo: &str) -> Option<Chain> {
```
[the remark this answers is not in the document]
Comment (gemcoder21) [reply gemstone/src/swapper/thorchain/chain.rs:109]: It should return different params in general

File: gemstone/src/swapper/thorchain/memo.rs [hunk 3]
```diff
@@ -0,0 +1,263 @@
+use super::chain::THORChainName;
+use primitives::Chain;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ThorchainMemo {
```
Comment (gemcoder21) [inline gemstone/src/swapper/thorchain/memo.rs:5]: simplify, we only need an asset

Comment (0xh3rman) [reply gemstone/src/swapper/thorchain/memo.rs:5]: yeah, makes sense
